src/main/java/hudson/plugins/ec2/EC2AbstractSlave.java

```diff
@@ -109,6 +112,10 @@
     /* The time at which we fetched the last instance data */
     protected transient long lastFetchTime;
 
+    /** Terminate was scheduled */
+    @Nonnull
+    protected transient ResettableCountDownLatch terminateScheduled = new ResettableCountDownLatch(1, false);
```

Comment:
@res0nance My java isn't very good, but I think when these objects are deserialized `ResettableCountDownLatch` is getting set to `null` by default and not a new object.

Reply:
I think you can fix this in `readResolve`